app/pages/sections/additional-services/order-item/controller.js

```diff
@@ -0,0 +1,187 @@
+import { postData, putData } from 'buying-catalogue-library';
+import { getContext, getErrorContext } from './contextCreator';
+import { logger } from '../../../../logger';
+import { getEndpoint } from '../../../../endpoints';
+import commonManifest from './commonManifest.json';
+import { getSelectedPriceManifest } from '../../../../helpers/controllers/manifestProvider';
+import { extractDate } from '../../../../helpers/controllers/extractDate';
+
+const formatFormData = ({ formData }) => ({
+  'deliveryDate-day': formData['deliveryDate-day']
+    ? formData['deliveryDate-day'].trim() : undefined,
+  'deliveryDate-month': formData['deliveryDate-month']
+    ? formData['deliveryDate-month'].trim() : undefined,
+  'deliveryDate-year': formData['deliveryDate-year']
+    ? formData['deliveryDate-year'].trim() : undefined,
+  quantity: formData.quantity
+    ? formData.quantity.trim() : undefined,
+  price: formData.price && formData.price.length > 0
+    ? formData.price.trim() : undefined,
+  selectEstimationPeriod: formData.selectEstimationPeriod
+    ? formData.selectEstimationPeriod.trim() : undefined,
+});
+
+export const getOrderItemContext = async ({
+  orderId,
+  orderItemId,
+  orderItemType,
+  itemName,
+  odsCode,
+  serviceRecipientName,
+  selectedPrice,
+  formData,
+}) => {
+  const selectedPriceManifest = getSelectedPriceManifest({
+    orderItemType,
+    provisioningType: selectedPrice.provisioningType,
+    type: selectedPrice.type,
+  });
+
+  return getContext({
+    commonManifest,
+    selectedPriceManifest,
+    orderId,
+    orderItemId,
+    itemName,
+    serviceRecipientName,
+    odsCode,
+    selectedPrice,
+    formData,
+  });
+};
+
+export const getOrderItemErrorPageContext = (params) => {
+  const formattedData = formatFormData({
+    formData: params.formData,
+  });
+
+  const selectedPriceManifest = getSelectedPriceManifest({
+    orderItemType: params.orderItemType,
+    provisioningType: params.selectedPrice.provisioningType,
+    type: params.selectedPrice.type,
+  });
+
+  const updatedParams = {
+    ...params,
+    commonManifest,
+    selectedPriceManifest,
+    formData: formattedData,
+  };
+
+  return getErrorContext(updatedParams);
+};
+
+const formatPostData = ({
+  selectedRecipientId,
+  serviceRecipientName,
+  selectedSolutionId,
```

Comment:
AdditionalService instead of Solution? The theme of this rename runs through the rest of the file.

Reply:
correct this is all part of the post and when we submit the form. I'll be doing that next. Told you it wasn't a nice PR

Reply:
Lol! No worries, just checking & understandable as we are in bit of a transition.